### ISSUE TITLE
Getting type hints in `Field.__set_name__` can cause `RuntimeError`

### ISSUE THREAD

**Issue by micuda** [smarie/python-pyfields#73]
Hi,
at first, thank You for the great library. =)

For code
```
from __future__ import annotations

from pyfields import field

class Foo:
  bar: Foo = field()
```

I got `RuntimeError: Error calling __set_name__ on 'NativeField' instance 'bar' in 'Foo'`. It is because class `Foo` does not exist yet and field calls `typing.get_type_hints` at line 299.
https://github.com/smarie/python-pyfields/blob/e19ab1285bf74a16678c33468eb8ec82dd6f0eab/pyfields/core.py#L293-L300

For now I just don't annotate attributes like this, but it is weird (and inconsistent) that some attributes have typing annotation and some don't. =)

I'm not sure where and how to fix it.

**Comment by smarie** [comment]
Thanks @micuda for reporting this issue ! I'll have a look asap

**Comment by smarie** [comment]
So after investigating a bit: `from __future__ import annotations` is a special line introduced in [PEP563](https://www.python.org/dev/peps/pep-0563/) in order to enable python 3.10 behaviour in earlier versions. However this behaviour becomes similar to what happens when you use string type hints. So I am able to reproduce the issue without the PEP563:

```python
from pyfields import field

class Foo:
  bar: 'Foo' = field()
```

I will fix this by enabling delayed type hint resolution.

**Comment by smarie** [comment]
Fixed in 1.3.0 (as well as field access from class #12, which was waiting a fix on pycharm side that is now available - I had to make the patch for them :) )
